examples/chef/common/clusters/valve-configuration-and-control/chef-valve-configuration-and-control-delegate-impl.cpp

```diff
@@ -0,0 +1,161 @@
+/*
+ *
+ *    Copyright (c) 2025 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include "chef-valve-configuration-and-control-delegate-impl.h"
+#include <app-common/zap-generated/attributes/Accessors.h>
+#include <app/clusters/valve-configuration-and-control-server/valve-configuration-and-control-server.h>
+#include <app/reporting/reporting.h>
+
+#ifdef MATTER_DM_PLUGIN_VALVE_CONFIGURATION_AND_CONTROL_SERVER
+
+using namespace chip::app::Clusters::ValveConfigurationAndControl;
+using namespace chip::app::Clusters;
+using namespace chip::app;
+
+static DelegateImpl * gValveConfigurationAndControlDelegate = nullptr;
+
+DataModel::Nullable<chip::Percent> DelegateImpl::HandleOpenValve(DataModel::Nullable<chip::Percent> level)
+{
+    if (!level.IsNull())
+    {
+        ChipLogProgress(DeviceLayer, "HandleOpenValve with level = %d", level.Value());
+    }
+    else
+    {
+        ChipLogProgress(DeviceLayer, "HandleOpenValve with level = NULL");
+    }
+    return level;
+}
+
+CHIP_ERROR DelegateImpl::HandleCloseValve()
+{
+    ChipLogProgress(DeviceLayer, "HandleCloseValve");
+    return CHIP_NO_ERROR;
+}
+
+void DelegateImpl::HandleRemainingDurationTick(uint32_t duration)
+{
+    ChipLogProgress(DeviceLayer, "HandleRemainingDurationTick with duration = %d", duration);
+}
+
+void ValveConfigurationAndControl::Shutdown()
+{
+    if (gValveConfigurationAndControlDelegate != nullptr)
+    {
+        delete gValveConfigurationAndControlDelegate;
+        gValveConfigurationAndControlDelegate = nullptr;
+    }
+}
+
+chip::Protocols::InteractionModel::Status
+chefValveConfigurationAndControlWriteCallback(chip::EndpointId endpointId, chip::ClusterId clusterId,
+                                              const EmberAfAttributeMetadata * attributeMetadata, uint8_t * buffer)
+{
+    if (endpointId != 1)
+    {
+        ChipLogError(DeviceLayer, "Endpoint %d not supported for ValveConfigurationAndControl cluster.", endpointId);
+        return chip::Protocols::InteractionModel::Status::UnsupportedEndpoint;
+    }
+
+    chip::AttributeId attributeId = attributeMetadata->attributeId;
+
+    switch (attributeId)
+    {
+    case chip::app::Clusters::ValveConfigurationAndControl::Attributes::RemainingDuration::Id: {
```

Comment:
```suggestion
    case Attributes::RemainingDuration::Id: {
```

You already use `chip::app::Clusters::ValveConfigurationAndControl`, so you can simplify the code throughout

Reply:
Done.